Core/Src/ThermistorMonitor.c

```diff
@@ -6,53 +6,149 @@
+//#define RES_COEFF_C -87493.00000000
+//#define RES_COEFF_D -9588114.00000000
+
+
 
 ThermistorData_Struct ThermistorData = { .thermistors = {} };
 
+void adcChangeMUX(int channel) {
+	ADC_ChannelConfTypeDef sConfig = {0};
+
+	// Selects new ADC channel for conversion
+	if (DEV_BOARD == 0) {
+		switch(channel) {
+
+		case 0:
+			sConfig.Channel = ADC_CHANNEL_6;
+			break;
+		case 1:
+			sConfig.Channel = ADC_CHANNEL_7;
+			break;
+		case 2:
+			sConfig.Channel = ADC_CHANNEL_8;
+			break;
+		case 3:
+			sConfig.Channel = ADC_CHANNEL_9;
+			break;
+		}
+	} else {
+		sConfig.Channel = ADC_CHANNEL_8;
+	}
+
+	sConfig.Rank = 1;
+	sConfig.SamplingTime = ADC_SAMPLETIME_28CYCLES_5;
+	if (HAL_ADC_ConfigChannel(&hadc1, &sConfig) != HAL_OK) {
+		Error_Handler();
+	}
+}
+
 void startThermistorMonitorTask() {
 	uint32_t tick = osKernelGetTickCount();
-	uint8_t counter = 0;
+	uint8_t select_line = 0;
+
+
 
 	while (1) {
-		// Counter Variable for MUX Select Lines
-        if (++counter == THERMISTORS_PER_CHANNEL) {
-            counter = 0;
-        }
 
+		// Resetting pins A, B, C; then setting the next binary sequence as per table 1:
 		HAL_GPIO_WritePin(GPIOA, 0b111000, GPIO_PIN_RESET);
-		HAL_GPIO_WritePin(GPIOA, counter<<3, GPIO_PIN_SET);
+		HAL_GPIO_WritePin(GPIOA, select_line<<3, GPIO_PIN_SET);
 
 		osDelay(100);
 
-        // Grab the temperature for all channels at once
-        for (uint8_t channel = 0; channel < CHANNEL_COUNT; channel++) {
+        // Grab the temperature for all modules at once
+        for (uint8_t cur_mux = 0; cur_mux < MUX_COUNT; cur_mux++) {
+        	// changes the MUX that the ADC polls
+        	adcChangeMUX(cur_mux);
             HAL_ADC_Start(&hadc1);
             HAL_ADC_PollForConversion(&hadc1, HAL_MAX_DELAY);
-            uint32_t divider_voltage = HAL_ADC_GetValue(&hadc1);
-
-            // Obtain Thermistor Resistance from Measured Voltage
-            float percent_of_divider_input_voltage = (ADC_RESOLUTION - (float)divider_voltage) / ADC_RESOLUTION;
-            float thermistor_resistance = DIVIDER_RESISTANCE * percent_of_divider_input_voltage;
-
-            // TODO: pick a better sentinel value
+            uint32_t divider_output = HAL_ADC_GetValue(&hadc1);
+            HAL_ADC_Stop(&hadc1);
+
+
+            DEBUG_PRINT("----------------------------\n");
+            DEBUG_PRINT("Divider ADC out: %d\n", divider_output);
+//
+//
+            // converts ADC units to volts
+            float divider_voltage = ((float) divider_output/(float) ADC_RESOLUTION * REFERENCE_VOLTAGE);
+//
+//
+            DEBUG_PRINT("Divider Voltage: %d.%d\n", (int)divider_voltage, (int) (divider_voltage * 1000) % 1000);
+//
+            // calculates resistance of the thermistor using rearrangement of voltage divider formula: https://ohmslawcalculator.com/voltage-divider-calculator
+            float thermistor_resistance = (divider_voltage * DIVIDER_RESISTANCE) / (REFERENCE_VOLTAGE - divider_voltage);
+//
```

Comment:
```suggestion
            // converts ADC units to volts
            float divider_voltage = ((float) divider_output/(float) ADC_RESOLUTION * REFERENCE_VOLTAGE);

            DEBUG_PRINT("Divider Voltage: %d.%d\n", (int)divider_voltage, (int) (divider_voltage * 1000) % 1000);
            // calculates resistance of the thermistor using rearrangement of voltage divider formula: https://ohmslawcalculator.com/voltage-divider-calculator
            float thermistor_resistance = (divider_voltage * DIVIDER_RESISTANCE) / (REFERENCE_VOLTAGE - divider_voltage);
```
Maybe just clean up some of your straggling comments